### PR TITLE
feat: implement get user by email - management API

### PR DESCRIPTION
**Pull request by danielgent**
Similar to get by Id

### CODE REVIEW

File: src/routes/tsoa/users-mgmt.ts [hunk 1]
```diff
@@ -48,4 +48,34 @@ export class UsersMgmtController extends Controller {
...
+    }
+
+    const user = env.userFactory.getInstanceByName(
+      getId(dbUser.tenantId, dbUser.email),
+    );
+
+    const userResult = user.getProfile.query();
+
+    return userResult;
+  }
```
Comment (danielgent) [inline src/routes/tsoa/users-mgmt.ts:80]: I'm testing all these in postman and on the demo repo - https://github.com/sesamyab/auth0-management-api-demo

image

File: src/routes/tsoa/users-mgmt.ts [hunk 2]
```diff
@@ -48,4 +48,34 @@ export class UsersMgmtController extends Controller {
...
+
+    if (!dbUser) {
+      throw new NotFoundError();
+    }
+
+    const user = env.userFactory.getInstanceByName(
+      getId(dbUser.tenantId, dbUser.email),
+    );
+
+    const userResult = user.getProfile.query();
```
Comment (markusahlstrand) [inline src/routes/tsoa/users-mgmt.ts:77]: This is tricky :) So the userResult is actually a promise, but it works as the function is async. I would either return directly on line 77 (my favorite..) or do an await so you get the actual profile.

File: src/routes/tsoa/users-mgmt.ts [hunk 3]
```diff
@@ -48,4 +48,34 @@ export class UsersMgmtController extends Controller {
...
     return userResult;
   }
+
+  @Get("users-by-email")
+  public async getUserByEmail(
+    @Request() request: RequestWithContext,
+    @Query("email") userEmail: string,
+    @Header("tenant-id") tenantId: string,
+  ): Promise<Profile> {
+    const { ctx } = request;
```
Comment (markusahlstrand) [inline src/routes/tsoa/users-mgmt.ts:58]: If we don't use the context we could replace these lines with:
`const { env } = request.ctx`

**Review by markusahlstrand — APPROVED** [review]
Looks good! Just a few style comments.

File: src/routes/tsoa/users-mgmt.ts (hunk 3 above)
Comment (danielgent) [reply src/routes/tsoa/users-mgmt.ts:58]: Done on future PR -> https://github.com/sesamyab/auth/pull/114/files#diff-5dc5f7c3ecf119f9d068a8003af576d25245dd803103d422fff43c7cf4680560R33

File: src/routes/tsoa/users-mgmt.ts [hunk 4]
```diff
@@ -48,4 +48,34 @@ export class UsersMgmtController extends Controller {
...
+    const { env } = ctx;
+
+    const db = getDb(env);
+    const dbUser = await db
+      .selectFrom("users")
+      .where("users.tenantId", "=", tenantId)
+      .where("users.email", "=", userEmail)
+      .selectAll()
```
[the remark this answers is not in the document]
Comment (danielgent) [reply src/routes/tsoa/users-mgmt.ts:66]: Done on this PR on all routes :+1:   https://github.com/sesamyab/auth/pull/113/files#diff-5dc5f7c3ecf119f9d068a8003af576d25245dd803103d422fff43c7cf4680560R41

File: src/routes/tsoa/users-mgmt.ts (hunk 2 above)
Comment (danielgent) [reply src/routes/tsoa/users-mgmt.ts:77]: All returns cleaned up here -> https://github.com/sesamyab/auth/pull/114/files#diff-5dc5f7c3ecf119f9d068a8003af576d25245dd803103d422fff43c7cf4680560R106

**Pull request merged**